File: frontend/admin/src/adm/overview/overview.ts
import { IOverviewPb } from "@proto/AdmOverviewProto.ts";
import { makeAutoObservable } from "mobx";

export class Overview {
  month = "";
  apiGet = 0;
  apiPost = 0;
  userCreated = 0;
  userDeleted = 0;
  roleCreated = 0;
  roleDeleted = 0;
  permissionCreated = 0;
  permissionDeleted = 0;

  constructor() {
    makeAutoObservable(this);
  }

  static ofPb(pb: IOverviewPb): Overview {
    const v = new Overview();
    v.month = pb.month;
    v.apiGet = pb.apiGet;
    v.apiPost = pb.apiPost;
    v.userCreated = pb.userCreated;
    v.userDeleted = pb.userDeleted;
    v.roleCreated = pb.roleCreated;
    v.roleDeleted = pb.roleDeleted;
    v.permissionCreated = pb.permissionCreated;
    v.permissionDeleted = pb.permissionDeleted;
    return v;
  }
}

File: service/src/main/java/io/github/jinganix/admin/starter/adm/overview/model/Overview.java
package io.github.jinganix.admin.starter.adm.overview.model;

import io.github.jinganix.admin.starter.helper.data.AbstractEntity;
import java.time.LocalDate;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@Getter
@Setter
@Accessors(chain = true)
public class Overview extends AbstractEntity {

  private Long id;

  private LocalDate month;

  private long apiGet;

  private long apiPost;

  private long userCreated;

  private long userDeleted;

  private long roleCreated;

  private long roleDeleted;

  private long permissionCreated;

  private long permissionDeleted;
}
